### PR TITLE
Add domain wide node selector

### PR DESCRIPTION
**Pull request by CarolynRountree**
Fixes Jira #356

Add node selector to domain wide server pod in advanced section

### CODE REVIEW

File: electron/app/locales/en/webui.json [hunk 1]
```diff
@@ -644,6 +644,8 @@
   "domain-design-username-header": "Username",
   "domain-design-password-header": "Password",
   "domain-design-secrets-cell-field-name": "Secrets table {{name}} field at row {{position}} with UID {{uid}}",
+  "domain-design-node-selector-label": "Kubernetes node selector",
```
Comment (rakillen) [inline electron/app/locales/en/webui.json:647]: Adjacent `*-label` entries are capitalized, should this be "Kubernetes Node Selector" ?

**Review by rakillen — APPROVED** [review]

File: webui/src/js/utils/k8s-domain-resource-generator.js [hunk 2]
```diff
@@ -247,7 +248,7 @@ define(['models/wkt-project', 'utils/k8s-domain-configmap-generator', 'js-yaml',
     }
 
     function getServerPodForCluster(cluster) {
-      const serverPod = _getServerPod(getJavaOptionsForCluster(cluster), getUserMemArgsForCluster(cluster), getKubernetesResourcesForCluster(cluster)) || {};
+      const serverPod = _getServerPod(getJavaOptionsForCluster(cluster), getUserMemArgsForCluster(cluster), getKubernetesResourcesForCluster(cluster), null) || {};
```
Comment (robertpatrick) [inline webui/src/js/utils/k8s-domain-resource-generator.js:251]: If you remove the ugly `null` argument, then the receiving function will simply get `undefined` for the argument value.  If it makes you feel better, you can add a default value for that argument in the function argument list.  That is a more JavaScript-way of doing this.

```
const serverPod = _getServerPod(getJavaOptionsForCluster(cluster), getUserMemArgsForCluster(cluster), getKubernetesResourcesForCluster(cluster)) || {};

...
function _getServerPod(javaOptions, userMemArgs, resources, nodeSelector = null) {
...

```

File: webui/src/js/utils/k8s-domain-resource-generator.js [hunk 3]
```diff
@@ -270,6 +271,12 @@ define(['models/wkt-project', 'utils/k8s-domain-configmap-generator', 'js-yaml',
         serverPod.resources = resources;
       }
 
+<<<<<<< HEAD
+      if (nodeSelector) {
+        serverPod.nodeSelector = { name: nodeSelector };
+      }
+=======
+>>>>>>> main
       return Object.keys(serverPod).length > 0 ? serverPod : null;
```
Comment (robertpatrick) [inline webui/src/js/utils/k8s-domain-resource-generator.js:280]: You do realize that you have committed code with merge conflict markers in it, right?

File: webui/src/js/views/domain-design-view.html [hunk 4]
```diff
@@ -468,6 +468,10 @@ <h6 class="wkt-subheading"><oj-bind-text value="[[labelMapper('domain-server-tit
                      validators="[[project.k8sDomain.memoryLimit.validators()]]"
                      help.instruction="[[labelMapper('memory-limit-help')]]">
       </oj-input-text>
+      <oj-input-text label-hint="[[labelMapper('node-selector-label')]]"
+                     value="{{project.k8sDomain.nodeSelector.observable}}" class="editable"
+                     help.instruction="[[labelMapper('node-selector-help')]]">
+      </oj-input-text>
```
Comment (robertpatrick) [inline webui/src/js/views/domain-design-view.html:474]: How is this supposed to work?

I would have assumed that the user can tell you both the label name and the label value that they want to use as the node selector?  It seems like you are assuming a label name but it isn't clear what that is...

**Review by robertpatrick — CHANGES_REQUESTED** [review]

**Comment by robertpatrick** [comment]
This will be added with the Verrazzano support that is coming in the next week or so.

**Pull request closed without merging**